pulsar/producer_partition.go

```diff
@@ -292,12 +292,12 @@ func (p *partitionProducer) grabCnx(assignedBrokerURL string) error {
 		}
 	}
 
-	res, err := p.client.rpcClient.Request(lr.LogicalAddr, lr.PhysicalAddr, id, pb.BaseCommand_PRODUCER, cmdProducer)
+	res, err := p.client.rpcClient.RequestOnCnx(cnx, id, pb.BaseCommand_PRODUCER, cmdProducer)
```

Comment:
Dont we need to handle the case when cnx is null, here?

Reply:
Perhaps we should add error handle in line 286, so we don't have to check for the `cnx` is nil.

Reply:
When the err is nil, the cnx should be non-null. This is a convention.

Reply:
Fixed.